data-prepper-plugins/aws-lambda/src/main/java/org/opensearch/dataprepper/plugins/lambda/processor/LambdaProcessor.java

```diff
@@ -324,9 +376,28 @@ void handleFailure(Throwable e, Buffer flushedBuffer, List<Record<Event>> result
             }
             resultRecords.add(record);
         }
-        LOG.error(NOISY, "Failed to process batch due to error: ", e);
     }
 
+    Buffer createBuffer(BufferFactory bufferFactory){
+        try {
+            return bufferFactory.getBuffer(lambdaAsyncClient, functionName, invocationType);
+        } catch (IOException e) {
+            LOG.error("Failed to create new buffer");
+            throw new RuntimeException(e);
+        }
+    }
+
+    private boolean isPayloadNullOrEmpty(SdkBytes payload) {
+        return payload == null || payload.asByteArray() == null || payload.asByteArray().length == 0;
+    }
+
+    private List<Event> parsePayload(SdkBytes payload, InputCodec responseCodec) throws IOException {
+        List<Event> parsedEvents = new ArrayList<>();
+        InputStream inputStream = PayloadValidator.validateAndGetInputStream(payload);
+        responseCodec.parse(inputStream, record -> parsedEvents.add(record.getData()));
+        LOG.info("Parsed successfully");
```

Comment:
make debug/trace. This would result in too much logging. The metrics should indicate this.